src/main/java/com/fjb/sunrise/controllers/AuthController.java

```diff
@@ -25,42 +30,76 @@ public class AuthController {
         modelAndView.setViewName(AUTH_VIEW);
-        modelAndView.addObject(LOGIN_ATTRIBUTE, new LoginRequest());
-        modelAndView.addObject(REGISTER_ATTRIBUTE, new RegisterRequest());
+        modelAndView.addObject(LOGIN_OBJECT, new LoginRequest());
+        modelAndView.addObject(REGISTER_OBJECT, new RegisterRequest());
         if (error != null) {
-            modelAndView.addObject(ERROR_MESSAGE, "Đăng nhập không thành công!");
+            modelAndView.addObject(ERROR_MESSAGE_OBJECT, "Đăng nhập không thành công!");
         }
         return modelAndView;
     }
 
     @GetMapping("/register")
-    public ModelAndView index() {
+    public ModelAndView indexRegister() {
         ModelAndView modelAndView = new ModelAndView();
         modelAndView.setViewName(AUTH_VIEW);
-        modelAndView.addObject(LOGIN_ATTRIBUTE, new LoginRequest());
-        modelAndView.addObject(REGISTER_ATTRIBUTE, new RegisterRequest());
+        modelAndView.addObject(LOGIN_OBJECT, new LoginRequest());
+        modelAndView.addObject(REGISTER_OBJECT, new RegisterRequest());
+        return modelAndView;
+    }
+
+    @GetMapping("/forgotPassword")
+    public ModelAndView indexForgotPassword() {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName(AUTH_VERIFICATION);
+        modelAndView.addObject(EMAIL_OBJECT, "");
+        modelAndView.addObject(PHONE_OBJECT, "");
+        return modelAndView;
+    }
+
+    @GetMapping("/inputCode")
+    public ModelAndView indexInputCode() {
+        ModelAndView modelAndView = new ModelAndView();
+        modelAndView.setViewName(AUTH_VERIFICATION_CODE);
+        modelAndView.addObject(CODE_OBJECT, "");
         return modelAndView;
     }
 
     @PostMapping("/register")
-    public ModelAndView doRegister(@ModelAttribute("register") RegisterRequest registerRequest) {
+    public ModelAndView doRegister(@ModelAttribute(REGISTER_OBJECT) RegisterRequest registerRequest) {
         //setup object for view
         ModelAndView modelAndView = new ModelAndView();
         modelAndView.setViewName(AUTH_VIEW);
-        modelAndView.addObject(LOGIN_ATTRIBUTE, new LoginRequest());
-        modelAndView.addObject(REGISTER_ATTRIBUTE, new RegisterRequest());
+        modelAndView.addObject(LOGIN_OBJECT, new LoginRequest());
+        modelAndView.addObject(REGISTER_OBJECT, new RegisterRequest());
 
         // implement register for user
         if (service.checkRegister(registerRequest)) {
             modelAndView.setViewName(AUTH_REDIRECT_LOGIN);
         } else {
-            modelAndView.addObject(ERROR_MESSAGE, "Đăng kí không thành công");
+            modelAndView.addObject(ERROR_MESSAGE_OBJECT, "Đăng kí không thành công");
         }
 
         return modelAndView;
     }
 
+    //this function is prepared in advance and will be rewritten later.
+    @PostMapping("/sendToEmail")
+    public ModelAndView doSendCodeToEmail(@ModelAttribute(EMAIL_OBJECT) String email) {
```

Comment:
Are there any dependencies required for sending an email or a phone message?

Reply:
I will add it immediately